stores/filestore.go

```diff
@@ -1139,15 +1149,35 @@ func NewFileStore(rootDir string, limits *StoreLimits, options ...FileStoreOptio
 		}
 	}
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 	// Create the recovered state to return
 	recoveredState = &RecoveredState{
 		Info:    serverInfo,
 		Clients: recoveredClients,
 		Subs:    recoveredSubs,
 	}
-	return fs, recoveredState, nil
+	return recoveredState, nil
+}
+
+// GetExclusiveLock implements the Store interface
+func (fs *FileStore) GetExclusiveLock() (bool, error) {
+	fs.Lock()
+	defer fs.Unlock()
+	if fs.lockFile != nil {
+		return true, nil
+	}
+	f, err := util.CreateLockFile(filepath.Join(fs.fm.rootDir, "ft.lck"))
```

Comment:
I should have made this a constant at the very least (especially that I use that name in a test). Will change that.